### PR TITLE
[framework] Responsive administration

### LINKED ISSUE #594: Broken design in administration with singledomain
### What is happening

If we have configured just one domain in administration there is some design bug
image

### Expected result

singledomain admin design will be fixed

### LINKED ISSUE #1297: In administration popups the headings are not centered
### What is happening
In administration popups: if there is a heading is not centered to a popup box, but centered between left end of box and closing cross.

This type of box appears in in-line edition when you delete the item that is assigned to some entity e.g. status of order.

Snímek obrazovky 2019-08-05 v 14 46 11

### Expected result
The heading should be centered to a box.

### LINKED ISSUE #1331: Admin>Product detail: when validation fails, there are no categories
### What is happening
1. Admin > Products: I click on create new product
2. I click "Create" in the fixed/floating footer
3. The page refreshed and I see Product detail with info "Please check the correctness of all data filled." (I didn't choose availability)
4. In "Assign to category" there are no categories available
Snímek obrazovky 2019-08-16 v 10 14 45

### Expected result
The categories should be there as was before validation failed.

### LINKED ISSUE #1657: admin: missing FE links and visibilities on domains on product detail page
### What is happening
During admin facelift in https://github.com/shopsys/shopsys/pull/906, the section with links to FE (along with visibilities on the given domains) was removed from admin product detail - it was placed under the left menu. We really miss this feature, it was widely used by us, as well as by customers. The box is displayed on hover in admin product list (in "visibility" column) so it should not be hard to put it on the detail page).

### Expected result
There is a section links to FE domains and displayed visibilities on the domains on admin product detail once again :slightly_smiling_face:

### PR DESCRIPTION
**Pull request by TomasGottvald**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Adds responsive style for redesigned administration.
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| closes #594, closes #1297, closes #1331, closes #1657 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: packages/framework/assets/styles/admin/libs/select2/select2-custom.less [hunk 1]
```diff
@@ -76,3 +76,18 @@
 .select2-results__option[aria-disabled=true] {
     display: none;
 }
+
+.select2 {
+    @media @query-mobile {
+        width: 100%!important;
```
Comment (dusanpirosik) [inline packages/framework/assets/styles/admin/libs/select2/select2-custom.less:82]: Add space before `!important` rule

File: packages/framework/src/Resources/views/Admin/Form/orderItems.html.twig [hunk 2]
```diff
@@ -2,82 +2,80 @@
...
+        <h2>{{ 'Order items'|trans }}</h2>
+        <div class="form-line form-line--no-padd">
+            <div class="table-touch-always">
+                {{ form_errors(form.itemsWithoutTransportAndPayment) }}
+                <div class="table-touch-always__in">
+                    <table class="table-main table-main--with-inline-form">
+                        <thead>
+                            <tr>
+                                <th><span class="padding-left-10">{{ 'Name'|trans }}</span></th>
                                     <th class="text-right">{{ 'Catalog number'|trans }}</th>
```
Comment (dusanpirosik) [inline packages/framework/src/Resources/views/Admin/Form/orderItems.html.twig:14]: Check indentation

File: packages/framework/src/Resources/views/Admin/Form/orderItems.html.twig [hunk 3]
```diff
@@ -2,82 +2,80 @@
...
-                                    <th class="text-right">{{ 'Amount'|trans }}</th>
-                                    <th class="text-right">{{ 'Unit'|trans }}</th>
-                                    <th class="text-right">{{ 'VAT rate (%)'|trans }}</th>
-                                    <th class="text-center">
+                                <th class="text-right"><span class="table-main__column-with-info">{{ 'Unit price including VAT'|trans }} ({{ currencySymbolByCurrencyId(order.currency.id) }})</span></th>
+                                <th class="text-right">{{ 'Amount'|trans }}</th>
+                                <th class="text-right">{{ 'Unit'|trans }}</th>
+                                <th class="text-right">{{ 'VAT rate (%)'|trans }}</th>
+                                <th class="text-center">
                                         <span class="display-inline-block min-width-80">
```
Comment (dusanpirosik) [inline packages/framework/src/Resources/views/Admin/Form/orderItems.html.twig:20]: Check indentation

File: packages/framework/src/Resources/views/Admin/Form/productParameterValue.html.twig [hunk 4]
```diff
@@ -1,53 +1,51 @@
...
                 {% do form.setRendered %}
             {% endfor %}
 
-            <tr class="js-parameters-empty-item table-form__row">
-                <td class="table-form__cell" colspan="4">{{ 'Add some parameters'|trans }}</td>
-            </tr>
-            </tbody>
-        </table>
+                <div class="js-parameters-empty-item table-params__row">
+                    <div class="table-params__cell" colspan="4">{{ 'Add some parameters'|trans }}</div>
```
Comment (dusanpirosik) [inline packages/framework/src/Resources/views/Admin/Form/productParameterValue.html.twig:45]: Atribute `colspan` in div element?

File: packages/framework/assets/styles/admin/components/form/choice.less [hunk 5]
```diff
@@ -17,4 +17,30 @@
...
+        display: inline-block;
+        padding: 7px 0;
+        position: relative;
+    }
+
+    &--inline-checkbox {
+        margin-top: 0;
+
+        label {
+
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/form/choice.less:31]: Thx fixed

File: packages/framework/assets/styles/admin/components/form/choice.less [hunk 6]
```diff
@@ -17,4 +17,30 @@
...
+                float: left;
+                position: relative;
+                bottom: -7px;
+            }
+
+            &:nth-child(2) {
+                display: inline-block;
+                margin-top: 0;
+                padding-left: 30px;
+
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/form/choice.less:42]: thx, fixed

File: packages/framework/assets/styles/admin/components/form/line.less [hunk 7]
```diff
@@ -60,20 +66,29 @@
 
     .form-group & {
         &:first-child {
-            padding-top: 20px;
+            padding-top: 10px;
+
+            @media @query-lg {
+                padding-top: 20px;
+            }
+
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/form/line.less:74]: thx, fixed

File: packages/framework/assets/styles/admin/components/form/line.less [hunk 8]
```diff
@@ -112,11 +140,30 @@
...
             margin-top: 10px;
+
+            &--multilanguage {
+                @media only screen and (min-width: @form-line-side-multilanguage-change-resolution){
+                    margin-left: @form-line-label-size;
+                }
+            }
+
+            &--multidomain {
+                @media only screen and (min-width: @form-line-side-multidomain-change-resolution){
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/form/line.less:159]: thx, fixed

File: packages/framework/assets/styles/admin/components/form/line.less [hunk 9]
```diff
@@ -126,17 +173,74 @@
...
+        }
+
+        &--info-baseline {
+            position: absolute;
+            left: calc(~"100% - 20px");
+            bottom: 22px;
+            width: 20px;
+            padding-right: 0;
+
+            @media only screen and (min-width: 392px){
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/form/line.less:235]: thx, fixed

File: packages/framework/assets/styles/admin/components/form/line.less [hunk 10]
```diff
@@ -173,6 +277,22 @@
 
         &--fixed-width-left {
             width: 40px;
+            right: 22px;
+            position: absolute;
+            text-align: right;
+
+            @media only screen and (min-width: 372px){
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/form/line.less:284]: thx, fixed

File: packages/framework/assets/styles/admin/components/form/line.less [hunk 11]
```diff
@@ -207,8 +337,18 @@
     }
 
     &__count {
-        display: table;
-        height: 32px;
+        display: inline-block;
+        margin-top: 5px;
+        width: 100%;
+
+        @media only screen and (min-width: @form-line-side-multidomain-change-resolution){
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/form/line.less:344]: thx, fixed

File: packages/framework/assets/styles/admin/components/form/price-heading.less [hunk 12]
```diff
@@ -1,13 +1,31 @@
...
-    font-size: 0;
-    text-align: right;
+    @media @query-xl {
+        .clearfix();
+        display: block;
+        margin-top: 3px;
+        padding: 10px 15px 10px @form-price-heading-gap;
+        text-align: right;
+
+        background: @bg-table;
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/form/price-heading.less:13]: thx, fixed

File: packages/framework/assets/styles/admin/components/in/grid-control.less [hunk 13]
```diff
@@ -1,49 +1,118 @@
...
-        width: 30%;
+        width: 100%;
+        margin-bottom: 5px;
 
-        font-size: 12px;
         color: @color-0;
+        font-size: 12px;
+
+        @media @query-lg {
+            width: calc(~"99.99% / 3");
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/in/grid-control.less:24]: global variable created, changed in all files

File: packages/framework/assets/styles/admin/components/in/grid-control.less [hunk 14]
```diff
@@ -1,49 +1,118 @@
...
+                    width: 42%;
+                }
             }
         }
 
         &--show {
             line-height: @in-grid-height;
 
-            .window-content & {
+            text-align: center;
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/in/grid-control.less:75]: thx, fixed

File: packages/framework/assets/styles/admin/components/in/image.less [hunk 15]
```diff
@@ -32,31 +32,72 @@
...
-            .center-image();
+        .in-product-visibility &,
+        .list-domain__item__image & {
+            height: 26px;
+        }
+
+        &--1 {
+            line-height: @in-image-height;
+
+            background: #b04454;
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/in/image.less:60]: attribute name fixed, I dont want to create variable, it is used only in one line. I will next conversation bcs there only one thing to change. Changed on these lines

File: packages/framework/assets/styles/admin/components/table/params.less [hunk 16]
```diff
@@ -0,0 +1,183 @@
...
+
+.table-params {
+    width: 100%;
+    border-collapse: collapse;
+    border-radius: @radius;
+
+    thead {
+        background: @color-f;
+        color: @color-base;
+        font-weight: bold;
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/table/params.less:15]: changed in all files

File: packages/framework/assets/styles/admin/components/table/params.less [hunk 17]
```diff
@@ -0,0 +1,183 @@
...
+    }
+
+    tbody {
+        background: @color-f;
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/table/params.less:19]: thx, fixed

File: packages/framework/assets/styles/admin/components/table/params.less [hunk 18]
```diff
@@ -0,0 +1,183 @@
...
+
+    &__wrap {
+        padding-top: 4px;
+        position: relative;
+    }
+
+    &__row {
+        display: flex;
+        flex-direction: column;
+        background: @color-f;
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/table/params.less:30]: thx, fixed

File: packages/framework/assets/styles/admin/layout/panel/side-menu.less [hunk 19]
```diff
@@ -171,15 +178,16 @@
...
-        }
-
         .open:not(.side-menu__item--active) &, .side-menu--collapsed .side-menu__item--active > & {
             .animated(0.1s, 0s);
             .fadeIn-scale-topLeft;
         }
 
+        &.hidden {
+            display: none;
+
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/layout/panel/side-menu.less:188]: thx, fixed

File: packages/framework/assets/styles/admin/libs/select2/select2-custom.less (hunk 1 above)
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/libs/select2/select2-custom.less:82]: fixed

File: packages/framework/src/Resources/views/Admin/Form/orderItems.html.twig (hunk 2 above)
Comment (TomasGottvald) [reply packages/framework/src/Resources/views/Admin/Form/orderItems.html.twig:14]: thx, fixed

File: packages/framework/src/Resources/views/Admin/Form/orderItems.html.twig (hunk 3 above)
Comment (TomasGottvald) [reply packages/framework/src/Resources/views/Admin/Form/orderItems.html.twig:20]: thx, fixed

File: packages/framework/src/Resources/views/Admin/Form/productParameterValue.html.twig (hunk 4 above)
Comment (TomasGottvald) [reply packages/framework/src/Resources/views/Admin/Form/productParameterValue.html.twig:45]: Probably it was table in past. Changed. Thx

File: packages/framework/src/Resources/views/Admin/Layout/layoutWithPanel.html.twig [hunk 20]
```diff
@@ -2,7 +2,7 @@
 
 {% block content %}
     <div class="web__container">
-        <div class="web__panel">
+        <div class="web__panel js-web-panel">
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/src/Resources/views/Admin/Layout/layoutWithPanel.html.twig:5]: Before my change - sidemenu was static. We just added collapse posibility and respo functions

File: packages/framework/src/Resources/views/Admin/Form/productParameterValue.html.twig [hunk 21]
```diff
@@ -1,53 +1,51 @@
...
+            </div>
+            <div class="js-parameters" data-prototype="{{ self.parameterRow(form.vars.prototype)|escape }}" data-index="{{ form|length }}">
 
             {% for key, parameter in form %}
                 {{ self.parameterRow(parameter, key) }}
             {% else %}
                 {% do form.setRendered %}
             {% endfor %}
 
-            <tr class="js-parameters-empty-item table-form__row">
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/src/Resources/views/Admin/Form/productParameterValue.html.twig:46]: fixed

File: packages/framework/src/Resources/views/Admin/Form/theme.html.twig [hunk 22]
```diff
@@ -889,14 +889,16 @@
                 </span>
             </label>
         {% endfor %}
-    </div>
-    {% if isSimple|default(false) == false %}
+        {% if isSimple|default(false) == false %}
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/src/Resources/views/Admin/Form/theme.html.twig:872]: I dont think so. I cant anything broken. There are conditions, and looks like ok

File: packages/framework/assets/styles/admin/components/table/touch-always.less [hunk 23]
```diff
@@ -5,6 +5,10 @@
     &__in {
         width: 232px;
 
+        @media @query-sm {
+
+        }
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/table/touch-always.less]: This is fixed.

File: packages/framework/assets/styles/admin/components/in/grid-control.less [hunk 24]
```diff
@@ -20,12 +21,21 @@
         font-size: 12px;
 
         @media @query-lg {
-            width: 30%;
+            width: calc(~"99.99% / 3");
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/in/grid-control.less:24]: It was changed to global variable. This is IE hack :))

File: packages/framework/assets/styles/admin/layout/panel/side-menu.less [hunk 25]
```diff
@@ -182,6 +182,7 @@
 
         &.hidden {
             display: none;
+
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/layout/panel/side-menu.less:188]: fixed

File: packages/framework/src/Resources/views/Admin/Form/theme.html.twig [hunk 26]
```diff
@@ -488,25 +488,25 @@
...
                 value="{{ productName|default(placeholder) }}"
                 disabled="disabled"
                 readonly="readonly"
             />
-        </span>
-        <span class="form-line__item">
-            <button
+            <span class="form-line__item form-line__item--button form-line__item--fixed-width-left">
+                <button
                 type="button"
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/src/Resources/views/Admin/Form/theme.html.twig:488]: fixed, thx

File: packages/framework/src/Resources/views/Admin/Menu/side_menu.html.twig [hunk 27]
```diff
@@ -46,6 +46,16 @@
     {% if item.hasChildren and options.depth is not same as(0) and item.displayChildren %}
         <ul{{ self.attributes(item.childrenAttributes, classes) }}>
             {{ block('children') }}
+            {% if item.name == "root" %}
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/src/Resources/views/Admin/Menu/side_menu.html.twig:49]: fixed

File: packages/framework/assets/styles/admin/components/in/image.less [hunk 28]
```diff
@@ -32,31 +32,72 @@
...
 
-        img {
-            .center-image();
+        .in-product-visibility &,
+        .list-domain__item__image & {
+            height: 26px;
+        }
+
+        &--1 {
+            line-height: @in-image-height;
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/in/image.less:58]: thx, fixed

File: packages/framework/assets/styles/admin/components/table/params.less [hunk 29]
```diff
@@ -0,0 +1,183 @@
...
+@table-param-cell-max-width: 315px;
+@table-param-cell-arrow-width: 40px;
+
+.table-params {
+    width: 100%;
+    border-collapse: collapse;
+    border-radius: @radius;
+
+    thead {
+        background: @color-f;
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/table/params.less]: fixed, thx

File: packages/framework/assets/styles/admin/components/table/params.less [hunk 30]
```diff
@@ -0,0 +1,183 @@
...
+    }
+
+    &__row {
+        display: flex;
+        flex-direction: column;
+        background: @color-f;
+        border-bottom: 1px solid @table-grid-line-color;
+
+        &--group {
+            background: @bg-info;
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/table/params.less]: fixed

File: packages/framework/assets/styles/admin/components/table/params.less [hunk 31]
```diff
@@ -0,0 +1,183 @@
...
+        }
+
+        thead & {
+            background: @bg-table;
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/table/params.less]: fixed

File: packages/framework/assets/styles/admin/components/table/params.less [hunk 32]
```diff
@@ -0,0 +1,183 @@
...
+
+        &__image {
+            float: left;
+            width: 46px;
+            height: 26px;
+            position: relative;
+            margin-right: 5px;
+
+            text-align: center;
+            background: @color-f;
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/table/params.less:138]: fixed

File: packages/framework/assets/styles/admin/components/table/params.less [hunk 33]
```diff
@@ -0,0 +1,183 @@
...
+                max-width: 100%;
+                max-height: 100%;
+                margin: auto;
+            }
+        }
+
+        &__group-name {
+            line-height: 26px;
+
+            font-weight: bold;
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/table/params.less]: changed in all files

File: packages/framework/assets/styles/admin/components/wrap/bar.less [hunk 34]
```diff
@@ -1,26 +1,65 @@
...
-        float: left;
-        margin-bottom: 0;
+        position: relative;
+        width: 100%;
+        margin-bottom: 5px;
 
         color: @color-title;
+
+        @media @query-md {
+            //width: calc(~"100% - @{wrap-bar-buttons-width}");
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/wrap/bar.less:26]: thx, removed

File: packages/framework/assets/styles/admin/core/form/btn.less [hunk 35]
```diff
@@ -99,9 +99,23 @@
             background-color: @btn-neutral-bg-hover;
         }
     }
+
+    &--chooser {
+        width: 42px;
+        padding: @btn-padding-vertical;
+
+        background: @btn-default-bg;
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/core/form/btn.less:107]: fixed

File: packages/framework/assets/styles/admin/core/form/btn.less [hunk 36]
```diff
@@ -99,9 +99,23 @@
...
+        color: @btn-default-color;
+
+        &:hover {
+            background: darken(@btn-default-bg, 10%);
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/core/form/btn.less:111]: fixed

File: packages/framework/assets/styles/admin/components/form/line.less [hunk 37]
```diff
@@ -112,11 +140,30 @@
...
 
+        @media @query-xl {
+            margin-left: @form-line-input-size;
+        }
+
         & + .form-line__side {
             margin-top: 10px;
+
+            &--multilanguage {
+                @media only screen and (min-width: @form-line-side-multilanguage-change-resolution){
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/styles/admin/components/form/line.less:152]: fixed

File: packages/framework/assets/js/admin/components/SideMenu.js [hunk 38]
```diff
@@ -25,6 +28,33 @@ export default class SideMenu {
...
+                $('body').addClass('active-menu');
+                this.$webPanel.addClass('open');
+            } else {
+                $('body').removeClass('active-menu');
+                this.$webPanel.removeClass('open');
+            }
+        });
+
+        // close left menu on overlay click
+        this.$sideMenuOverlay.click(event => {
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/js/admin/components/SideMenu.js:44]: fixed

File: packages/framework/assets/js/admin/components/SideMenu.js [hunk 39]
```diff
@@ -25,6 +28,33 @@ export default class SideMenu {
...
+            $('body').removeClass('active-menu');
+            this.$webPanel.removeClass('open');
+        });
+
+        // show/hide left menu on desktop
+        this.$sideMenuCollapseButton.click(event => {
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/js/admin/components/SideMenu.js:50]: fixed

File: packages/framework/assets/js/admin/components/SideMenu.js [hunk 40]
```diff
@@ -3,19 +3,23 @@ import Register from '../../common/utils/Register';
...
-        this.$items.click(function () {
-            if ($(this).hasClass('open')) {
-                $(this).parents(this.$elementPanelClass).removeClass('open');
+        const _this = this;
+
+        this.$items.click( event => {
+            if($(event.currentTarget).hasClass('open')){
+                $(event.currentTarget).filterAllNodes('.js-side-menu-submenu').addClass('hidden');
+                $(event.currentTarget).removeClass('open');
+                _this.$webPanel.removeClass('open');
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply packages/framework/assets/js/admin/components/SideMenu.js:18]: fixed

**Pull request merged**